### PR TITLE
feat: observations list as a tab

### PR DESCRIPTION
**Pull request by ErikSin**
Adds observation list as tab in the home screen.

image

image

### CODE REVIEW

File: src/frontend/Navigation/types.ts [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
   focused: boolean;
   color: string;
 }
 
-export enum TabName {
-  Map = 'Map',
-  Camera = 'Camera',
-  Tracking = 'Tracking',
-}
+export type TabName = keyof HomeTabsList;
```
Comment (ErikSin) [inline src/frontend/Navigation/types.ts:10]: I replaced this `enum` with a `type`

**Review by achou11 — APPROVED** [review]
Changes look generally good. Couple of questions:

1. How do we get to the settings screen now that the button has been removed from the ObservationsList screen?

2. Couple of small issues with the tab icon (i'm okay with them being addressed in a follow-up): 

- the highlighting when active looks off. guessing that's an issue with the asset and will need to replace it.
- size is a little bigger than the other tab icons

File: src/frontend/screens/ObservationsList/ObservationListHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pressColor?: string | undefined;
+  pressOpacity?: number | undefined;
+  labelVisible?: boolean | undefined;
+}) => {
+  const {goBack} = useNavigationFromHomeTabs();
+
+  return (
+    <TouchableOpacity>
+      <CustomHeaderLeft
+        style={{marginLeft: 10}}
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/screens/ObservationsList/ObservationListHeader.tsx]: https://github.com/digidem/comapeo-mobile/pull/309/commits/b9a170b86dfa06debedc03d039cb57878b757fc8

File: src/frontend/Navigation/ScreenGroups/AppScreens.tsx [hunk 3]
```diff
@@ -63,20 +63,17 @@ import {
   SyncScreen,
   createNavigationOptions as createSyncNavOptions,
 } from '../../screens/Sync';
+// import {ObservationListIcon} from '../../sharedComponents/icons';
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/Navigation/ScreenGroups/AppScreens.tsx]: https://github.com/digidem/comapeo-mobile/pull/309/commits/b9a170b86dfa06debedc03d039cb57878b757fc8

File: src/frontend/screens/ObservationsList/ObservationListHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import {CustomHeaderLeft} from '../../sharedComponents/CustomHeaderLeft';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {useNavigationFromHomeTabs} from '../../hooks/useNavigationWithTypes';
+
+export const ObservationListHeader = (props: {
+  tintColor?: string | undefined;
+  pressColor?: string | undefined;
+  pressOpacity?: number | undefined;
+  labelVisible?: boolean | undefined;
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/screens/ObservationsList/ObservationListHeader.tsx]: https://github.com/digidem/comapeo-mobile/pull/309/commits/b9a170b86dfa06debedc03d039cb57878b757fc8

File: src/frontend/screens/ObservationsList/index.tsx [hunk 5]
```diff
@@ -1,13 +1,13 @@
 import * as React from 'react';
 import {View, FlatList, Dimensions, StyleSheet} from 'react-native';
-import {defineMessages} from 'react-intl';
+// import {defineMessages} from 'react-intl';
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/screens/ObservationsList/index.tsx]: https://github.com/digidem/comapeo-mobile/pull/309/commits/b9a170b86dfa06debedc03d039cb57878b757fc8

File: src/frontend/hooks/useCurrentTab.ts [hunk 6]
```diff
@@ -3,13 +3,27 @@ import {useGPSModalContext} from '../contexts/GPSModalContext';
 import {useTabNavigationStore} from './useTabNavigationStore.ts';
 import {TabName} from '../Navigation/types';
 
+const tabNames: TabName[] = ['Map', 'Camera', 'Tracking', 'ObservationList'];
+
+// Function to check if a given string is a valid TabName
+const isTabName = (name: string | undefined): name is TabName => {
+  return tabNames.includes(name as TabName);
+};
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/hooks/useCurrentTab.ts]: https://github.com/digidem/comapeo-mobile/pull/309/commits/b9a170b86dfa06debedc03d039cb57878b757fc8

File: src/frontend/hooks/useCurrentTab.ts [hunk 7]
```diff
@@ -3,13 +3,27 @@ import {useGPSModalContext} from '../contexts/GPSModalContext';
 import {useTabNavigationStore} from './useTabNavigationStore.ts';
 import {TabName} from '../Navigation/types';
 
+const tabNames: TabName[] = ['Map', 'Camera', 'Tracking', 'ObservationList'];
+
+// Function to check if a given string is a valid TabName
```
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/hooks/useCurrentTab.ts]: https://github.com/digidem/comapeo-mobile/pull/309/commits/b9a170b86dfa06debedc03d039cb57878b757fc8

File: src/frontend/screens/ObservationsList/ObservationListHeader.tsx
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/screens/ObservationsList/ObservationListHeader.tsx:1]: https://github.com/digidem/comapeo-mobile/pull/309/commits/9e4520986807c5881816e7cb4a15176d0e80468d

File: src/frontend/screens/ObservationsList/index.tsx
[the remark this answers is not in the document]
Comment (ErikSin) [reply src/frontend/screens/ObservationsList/index.tsx:1]: https://github.com/digidem/comapeo-mobile/pull/309/commits/9e4520986807c5881816e7cb4a15176d0e80468d

File: src/frontend/screens/ObservationsList/index.tsx [hunk 8]
```diff
@@ -91,6 +89,22 @@ export const ObservationsList: NativeNavigationComponent<'ObservationList'> = ({
   );
 };
 
+export function CreateNavigationOptions(): BottomTabNavigationOptions {
```
Comment (achou11) [inline src/frontend/screens/ObservationsList/index.tsx:92]: nit: to be consistent (also this is a normal function, not a component to being uppercase isn't strictly necessary)

```suggestion
export function createNavigationOptions(): BottomTabNavigationOptions {
```

Comment (achou11) [reply src/frontend/screens/ObservationsList/index.tsx:92]: kind of surprised doing `useIntl()` here works. i would suggest following the approach that existing implementations use though

Comment (ErikSin) [reply src/frontend/screens/ObservationsList/index.tsx:92]: this is actually making me realize that i am using a hook in it...

Comment (ErikSin) [reply src/frontend/screens/ObservationsList/index.tsx]: https://github.com/digidem/comapeo-mobile/pull/309/commits/56f1954b6b4c573fbd9b1d31554ea06a2969ec46

**Comment by ErikSin** [comment]
> 1. How do we get to the settings screen now that the button has been removed from the ObservationsList screen?

For posterity, the next PR I will be working on introduces the setting drawer which will replace the settings screen.

**Pull request merged**